enterprise/cypher/cypher/src/main/scala/org/neo4j/cypher/internal/spi/v3_2/codegen/GeneratedMethodStructure.scala

```diff
@@ -1357,7 +1357,9 @@ class GeneratedMethodStructure(val fields: Fields, val generator: CodeBlock, aux
   override def indexSeek(iterVar: String, descriptorVar: String, value: Expression, codeGenType: CodeGenType) = {
     val predicate = generator.declare(typeRef[IndexQuery], s"${iterVar}Query")
     val local = generator.declare(typeRef[PrimitiveLongIterator], iterVar)
-    val boxedValue = if (codeGenType.isPrimitive) Expression.box(value) else value
+    val boxedValue =
```

Comment:
Confusingly we no longer use this method, when reading from the index it doesn't matter what kind of index it is. I removed the unused method in a new commit.